### ISSUE TITLE
Autocomplete cursor movement

### ISSUE THREAD

**Issue by yurivish** [observablehq/feedback#327]
**Describe the bug**
Autocomplete does not move the cursor to the end of the inserted word.

**To Reproduce**
See https://observablehq.com/d/f54dc01b4462d120

**Expected behavior**

![image](https://user-images.githubusercontent.com/114388/147789225-8ad69050-ee45-4528-b80e-fddb82ffcc10.png)

**Screenshots**
![image](https://user-images.githubusercontent.com/114388/147789193-18a13d14-1ba6-4bb5-adf3-eba298cf2e75.png)

![image](https://user-images.githubusercontent.com/114388/147789202-0bde6d38-77d6-4629-9af6-2f358bacfdcd.png)

**Desktop (please complete the following information):**
macOS 10.15.7, Chrome Version 96.0.4664.110 (Official Build) (x86_64)

**Comment by hellonearthis** [comment]
Yes, it would be nice if the autocomplete replaced all the existing function text with the selected auto complete text.

**Comment by yurivish** [comment]
Is this a feature request? I thought it was a bug, since the cursor ends up in an unexpected place after autocompletion.

**Comment by hellonearthis** [comment]
I think it's a feature that looks like a bug.  I don't think patching the cursor position alone would be best solution and changing the way the autocomplete performs would fix the bug.

**Comment by mootari** [comment]
@hellonearthis This issue is definitely a bug. If you want to propose a different behavior, I recommend filing a separate issue for that.

**Comment by mootari** [comment]
The positioning fails for object properties, but only when an option is selected immediately after you type ".".

Fails:
1. type "DOM."
2. select an option

Works:
1. type "DOM."
2. hit Escape to close modal
3. type "c" and select a suggestion
